app/src/views/private/components/render-template.vue

```diff
@@ -27,140 +27,121 @@ const templateEl = ref<HTMLElement>();
 
 const regex = /({{.*?}})/g;
 
+const getNestedValues = (data: any, path: string) => {
+	const pathParts = path.split('.');
+	let currentData = data;
+
+	for (const part of pathParts) {
+		if (Array.isArray(currentData)) {
+			currentData = flatMap(currentData.map((item) => get(item, part)));
+		} else {
+			currentData = get(currentData, part);
+		}
+	}
+
+	return Array.isArray(currentData) ? currentData : [currentData];
+};
+
 const parts = computed(() =>
 	props.template
 		.split(regex)
 		.filter((p) => p)
 		.map((part) => {
-			if (part.startsWith('{{') === false) return part;
+			if (part.startsWith('{{') === false) return [part];
 
 			const fieldKey = part.replace(/{{/g, '').replace(/}}/g, '').trim();
-			const fieldKeyBefore = fieldKey.split('.').slice(0, -1).join('.');
-			const fieldKeyAfter = fieldKey.split('.').slice(-1)[0];
 
-			// Try getting the value from the item, return some question marks if it doesn't exist
-			const value = get(props.item, fieldKeyBefore);
-
-			return Array.isArray(value) ? handleArray(fieldKeyBefore, fieldKeyAfter) : handleObject(fieldKey);
+			// Try getting the value from the item
+			const value = getNestedValues(props.item, fieldKey);
+
+			let field: Field | null = props.fields?.find((field) => field.field === fieldKey) ?? null;
+
+			if (props.collection) {
+				field = fieldsStore.getField(props.collection, fieldKey);
+			}
+
+			if (!field) return [value];
+
+			const component = field?.meta?.display || getDefaultDisplayForType(field.type);
+			const options = field?.meta?.display_options;
+
+			// No need to render the empty display overhead in this case
+			if (component === 'raw') return [value];
+
+			const displayInfo = useExtension(
+				'display',
+				computed(() => component ?? null),
+			);
+
+			if (!displayInfo.value) return [value];
```

Comment:
Just for my understanding:
Why do we return `[value]` instead of `value` since `getNestedValues()` already returns an array? Then below if there are displays, we return an array of objects, yes? And at the bottom we map the values in `.map((p) => p ?? null),`, where `p` will always be an array, so it can't be `null`, right? And finally in the template, we have two loops, because we are wrapping these values in an array, right?
Does that make sense?